### PR TITLE
Set attach type during load for BPF_PROG_TYPE_CGROUP_SKB

### PR DESCRIPTION
**Pull request by nak3**
As per title, this patch sets `expected_attach_type` during load.

Fix https://github.com/aya-rs/aya/issues/262

### CODE REVIEW

**Review by vadorovsky — APPROVED** [review]

**Comment by dave-tucker** [comment]
@nak3 Thanks for the PR!

Does the current behaviour cause an issue (other than misleading strace output)?
Since `CgroupSkb` programs can be attached in either direction, and [you have to provide a direction on `attach` anyway,]( 
https://github.com/aya-rs/aya/blob/2fca4aee4e98b78c928be0c7aeb4c2166d93548a/aya/src/programs/cgroup_skb.rs#L89), I guess this change is just cosmetic?

I'm not against the change at all... I just want to understand the motivation.

File: aya/src/programs/cgroup_skb.rs [hunk 1]
```diff
@@ -62,6 +62,11 @@ pub struct CgroupSkb {
 impl CgroupSkb {
     /// Loads the program inside the kernel.
     pub fn load(&mut self) -> Result<(), ProgramError> {
+        self.data.expected_attach_type = match self.expected_attach_type {
```
Comment (alessandrod) [inline aya/src/programs/cgroup_skb.rs:65]: this should use map()

**Comment by nak3** [comment]
Hi @dave-tucker

Yes, one very small issue is caused. (Sorry I should explain it in the linked issue.)
As [Kernel code](https://github.com/torvalds/linux/commit/5cf1e91456301f8c4f6bbc63ff76cff12f92f31b) allows only Egress to use the return values (2 or 3), currently attach type "Egress" causes the following error:

```rust
#[cgroup_skb(name="cgroup_skb_tmp", attach="egress")]
pub fn cgroup_skb_tmp(ctx: SkBuffContext) -> i32 {
    3
}
```

this (return `3`) causes the following error:

```
Error: the BPF_PROG_LOAD syscall failed. Verifier output: func#0 @0
0: R1=ctx(id=0,off=0,imm=0) R10=fp0
0: (b7) r0 = 3                        ; R0_w=inv3
1: (95) exit
At program exit the register R0 has value (0x3; 0x0) should have been in (0x0; 0x1)
verification time 10 usec
stack depth 0
processed 2 insns (limit 1000000) max_states_per_insn 0 total_states 0 peak_states 0 mark_read 0


Caused by:
    Invalid argument (os error 22)
```

After applied this patch, the return value `2` or `3` for Egress is allowed.

File: aya/src/programs/cgroup_skb.rs (hunk 1 above)
Comment (nak3) [reply aya/src/programs/cgroup_skb.rs]: Thank you! updated.

**Review by alessandrod — APPROVED** [review]

**Pull request merged**